internal/server/server_test.go

```diff
@@ -0,0 +1,392 @@
+	m.toolNames = append(m.toolNames, toolName)
+
+	// Categorize tools
+	azureToolPrefixes := []string{"az_", "azure_", "get_aks_", "list_detectors", "run_detector"}
+	k8sToolPrefixes := []string{"kubectl_", "k8s_", "helm", "cilium", "inspektor"}
+
+	isAzureTool := false
+	for _, prefix := range azureToolPrefixes {
+		if containsPrefix(toolName, prefix) {
+			m.azureTools++
+			isAzureTool = true
+			break
+		}
+	}
+
+	if !isAzureTool {
+		for _, prefix := range k8sToolPrefixes {
+			if containsPrefix(toolName, prefix) {
+				m.k8sTools++
+				break
+			}
+		}
+	}
+}
+
+// GetCounts returns the tool counts
+func (m *MockToolCounter) GetCounts() (azureTools, k8sTools int) {
+	return m.azureTools, m.k8sTools
+}
+
+// GetToolNames returns all registered tool names
+func (m *MockToolCounter) GetToolNames() []string {
+	return m.toolNames
+}
+
+// TestService tests the service initialization and expected tool counts
+func TestService(t *testing.T) {
+	// Set environment variables for testing to avoid Azure auth issues
+	_ = os.Setenv("AZURE_TENANT_ID", "test-tenant")
+	_ = os.Setenv("AZURE_CLIENT_ID", "test-client")
+	_ = os.Setenv("AZURE_CLIENT_SECRET", "test-secret")
+	_ = os.Setenv("AZURE_SUBSCRIPTION_ID", "test-subscription")
+	defer func() {
+		_ = os.Unsetenv("AZURE_TENANT_ID")
+		_ = os.Unsetenv("AZURE_CLIENT_ID")
+		_ = os.Unsetenv("AZURE_CLIENT_SECRET")
+		_ = os.Unsetenv("AZURE_SUBSCRIPTION_ID")
+	}()
+
+	tests := []struct {
+		name               string
+		accessLevel        string
+		additionalTools    map[string]bool
+		expectedAzureTools int
+		expectedK8sTools   int
+		description        string
+	}{
+		{
+			name:               "ReadOnly_NoOptional",
+			accessLevel:        "readonly",
+			additionalTools:    map[string]bool{},
+			expectedAzureTools: 7, // AKS Ops + Monitoring + Fleet + Network + Compute (VMSS Info only) + Detectors (3) + Advisor
+			expectedK8sTools:   0, // Will be calculated based on kubectl tools for readonly
+			description:        "Readonly access with no optional tools",
+		},
+		{
+			name:               "ReadWrite_NoOptional",
+			accessLevel:        "readwrite",
+			additionalTools:    map[string]bool{},
+			expectedAzureTools: 8, // Same as readonly + 1 read-write VMSS command
+			expectedK8sTools:   0, // Will be calculated based on kubectl tools for readwrite
+			description:        "Readwrite access with no optional tools",
+		},
+		{
+			name:               "Admin_NoOptional",
+			accessLevel:        "admin",
+			additionalTools:    map[string]bool{},
+			expectedAzureTools: 8, // Same as readwrite (no admin VMSS commands currently)
+			expectedK8sTools:   0, // Will be calculated based on kubectl tools for admin
+			description:        "Admin access with no optional tools",
+		},
+		{
+			name:        "ReadOnly_AllOptional",
+			accessLevel: "readonly",
+			additionalTools: map[string]bool{
+				"helm":             true,
+				"cilium":           true,
+				"inspektor-gadget": true,
```

Comment:
The test needs an update we it is using the previous logic of additional tools for IG? Also can you please verify here that IG tool is registered by default similar to other tools?